python-avd/pyavd/_cv/client/__init__.py

```diff
@@ -182,3 +374,13 @@ def __init__(
         self._username = username
         self._password = password
         self._verify_certs = verify_certs
+        self._proxy_host = proxy_host
+        self._proxy_port = proxy_port
+        self._proxy_username = proxy_username
+        self._proxy_password = proxy_password
+        self._proxy_ssl = proxy_ssl
+        self._proxy_verify_certs = proxy_verify_certs
+        self._proxy_unix_socket_forwarder = proxy_unix_socket_forwarder
+        self._custom_ca_path = custom_ca_path
+        self._proxy_forwarder = None
+        self._proxy_socket_path = None
```

Comment:
I think having a separate class to handle the low level connection part would be beneficial.